[SPARK-23036][SQL][TEST] Add withGlobalTempView for testing

xubo245 opened the pull request:
## What changes were proposed in this pull request?

Add withGlobalTempView when create global temp view, like withTempView and withView.
And correct some improper usage.
Please see jira.
There are other similar place like that. I will fix it if community need. Please confirm it. 
## How was this patch tested?

no new test.

AmplabJenkins commented:
Can one of the admins verify this patch?

dongjoon-hyun commented on sql/core/src/test/scala/org/apache/spark/sql/execution/GlobalTempViewSuite.scala:
```diff
@@ -140,8 +140,8 @@ class GlobalTempViewSuite extends QueryTest with SharedSQLContext {
-      spark.catalog.dropTempView("v1")
-      spark.catalog.dropGlobalTempView("v2")
+      spark.catalog.dropGlobalTempView("v1")
+      spark.catalog.dropTempView("v2")
```
Hi, @xubo245 .
Could you split the bug and new improvement into two separate PR?
Then, your PR will get reviews more easily.

xubo245 replied:
Ok, done. Please review: https://github.com/apache/spark/pull/20250

dongjoon-hyun commented:
BTW, please put '[SQL]' into your title.
Then, your PR will be listed under SQL category here.
- https://spark-prs.appspot.com/open-prs

gatorsmile commented:
@xubo245 This is only for TEST? Then, add the tag [TEST] in your title too.

gatorsmile commented:
ok to test

SparkQA commented:
**[Test build #85976 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/85976/testReport)** for PR 20228 at commit [`fffd109`](https://github.com/apache/spark/commit/fffd109e8c084f9a4d63840bf761364f1ede5dc9).

SparkQA commented:
**[Test build #85976 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/85976/testReport)** for PR 20228 at commit [`fffd109`](https://github.com/apache/spark/commit/fffd109e8c084f9a4d63840bf761364f1ede5dc9).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/85976/
Test PASSed.

xubo245 committed:
[SPARK-23036] Add withGlobalTempView for testing

    Add withGlobalTempView when create global temp view, like withTempView and withView.

SparkQA commented:
**[Test build #86041 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/86041/testReport)** for PR 20228 at commit [`198f1ff`](https://github.com/apache/spark/commit/198f1ff54e88bf507692cdbc04669b5ee44f1678).

xubo245 commented:
ok, done, I  put '[SQL]' into title.  @dongjoon-hyun

xubo245 commented:
@gatorsmile Sure. This is only for TEST. Done , I put '[SQL]' into title too

SparkQA commented:
**[Test build #86041 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/86041/testReport)** for PR 20228 at commit [`198f1ff`](https://github.com/apache/spark/commit/198f1ff54e88bf507692cdbc04669b5ee44f1678).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/86041/
Test FAILed.

dongjoon-hyun commented:
Retest this please.

SparkQA commented:
**[Test build #86052 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/86052/testReport)** for PR 20228 at commit [`198f1ff`](https://github.com/apache/spark/commit/198f1ff54e88bf507692cdbc04669b5ee44f1678).

SparkQA commented:
**[Test build #86052 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/86052/testReport)** for PR 20228 at commit [`198f1ff`](https://github.com/apache/spark/commit/198f1ff54e88bf507692cdbc04669b5ee44f1678).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/86052/
Test PASSed.

gatorsmile commented on sql/core/src/test/scala/org/apache/spark/sql/execution/GlobalTempViewSuite.scala:
```diff
@@ -36,7 +36,7 @@ class GlobalTempViewSuite extends QueryTest with SharedSQLContext {
   test("basic semantic") {
     val expectedErrorMsg = "not found"
-    try {
+    withGlobalTempView("src") {
```
This is wrong. Do you know why?

gatorsmile commented:
LGTM

gatorsmile commented:
Thanks! Merged to master/2.3
